src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -382,6 +367,37 @@ public void copyResources(Resource[] resources, String prefix, String targetDire
         }
     }
 
+    /**
+     * Replaces filenames where the template name differs from the name the file should have in the repository.
+     *
+     * @param filePath The path to a file.
+     * @return The path with replacements applied where necessary.
+     */
+    private String applySpecialFilenameReplacements(final String filePath) {
+        String resultFilePath = filePath;
+
+        for (final Map.Entry<String, String> replacementDirective : FILENAME_REPLACEMENTS.entrySet()) {
+            String oldName = replacementDirective.getKey();
+            String newName = replacementDirective.getValue();
+
+            if (resultFilePath.endsWith(replacementDirective.getKey())) {
+                resultFilePath = resultFilePath.replace(oldName, newName);
+            }
+        }
+
+        return resultFilePath;
```

Comment:
Is it really necessary to iterate over all entries for every file? Is it even possible that two replacements could be done for one file? Or could we also do the following:

```suggestion
        return FILENAME_REPLACEMENTS.entrySet().stream()
        .findAny(entry -> filePath.endsWith(entry.getKey()))
        .map(entry -> filePath.replace(entry.getKey(), entry.getValue()))
        .orElse(filePath);
```

Reply:
I also thought about that but IMO this makes the code only harder to understand. In addition, `findAny` does not take any argument so you would have to use `filter` before which makes it even more cumbersome.

Reply:
Good point. I now implemented it by just adding a `break` inside the `if`-block to exit after the first successful replacement.